Code Examples/week4/bridge-vue/README.md

```diff
@@ -1232,3 +1232,152 @@ These techniques can be used in your own applications and components.  Also, now
+- apply Semantic middleware to Vue engine
+- rework BridgeMenu component
+
+To install dependencies:
+```
+npm install semantic-ui-vue semantic-ui-css --save
+```
+
+Then, to make Vue understand Semantic's custom classes and elements, stylesheet and JS module files are imported into `main.js`:
+```
+import SuiVue from 'semantic-ui-vue';
+Vue.use(SuiVue);
+import 'semantic-ui-css/semantic.min.css';
+```
+
+<strong>Template markup</strong> of `BridgeMenu.vue` component is modified accordingly. 
+
+First, outer container `<div class='main'>` is now <em>classified</em>, instead of <em>identified</em>,  as `menu`, for the reason of CSS rules to be reused. 
+
+In this markup, conditional directives `v-if`, `v-else` replace `v-bind`ing of class-attributes (i.e. `active`) that Semantic uses for the sake of readability. 
+
+Lastly, when menu-items are being deployed into the listing, the source array is changed to a new stateful property `this.showBridges`, that is a filtered version of `this.bridges`. 
+
+```angular2html
+<template>
+    <div class="menu">
+        <!--Search Field-->
+        <div style="width:100%;" class="ui icon input">
+            <input placeholder="Name a bridge..." v-model="search" />
+            <i class="search icon"></i>
+        </div>
+        <div class="menu">
+            <!--Loader-->
+            <div v-if="status.loading" class="menu ui segment inverted">
+                <div class="ui active dimmer">
+                    <div class="ui text loader">Loading bridges...</div>
+                </div>
+            </div>
+            <!--Error Modal-->
+            <div v-if="status.errored">
+                <div class="ui warning message">
+                    <div class="header">
+                        Error: Data was not loaded.
+                    </div>
+                    Hint: Check API endpoint!
+                </div>
+            </div>
+            <!--BridgeListing-->
+            <div v-else>
+                <menu-item
+                        v-for="bridge of shownBridges" <!--filtered array-->
+                        :key="bridge.id"
+                        :bridge="bridge"
+                        @click="bridgeSelected"
+                />
+            </div>
+        </div>
+    </div>
+</template>
+```
+
+Small change to `<style>`, to suit multiple 'menu'-like elements:
+```angular2html
+<style scoped>
+    .menu {
+        height: 100%;
+    }
+</style>
+```
+
+En fin, under `<script>`, we need to declare 2 new variables to the state: `search` and `shownBridges`. Former will be synced (`v-model`) with SearchBar, while latter will be dynamically updated with filtered array of bridges. 
+
+To filter out `shownBridges`, `search` property will be <em>watched</em>, so that, whence new value is entered, the listing is rendered. <em>Notably</em>, original array stays in place, even more so, it is going to be automatically updated in case of fresh data downloaded from backend.
```

Comment:
First, a question: why not just use a `computed` value here for the filtered bridges, which depends on both `this.search` and `this.bridges`?  It would return the filtered array of bridges when either/both updates.  I think that's more idiomatic of Vue.

Reply:
@humphd, computed values permit no mutation from inside. `filteredBridges` may be removed from state, however,` v-for of computed` is malformed.
How would you implement this?

Reply:
` v-for="(bridge, index) of filteredBridges"` and rename your `computed` property to `filteredBridges`